lib/ably/rest/client.rb

```diff
@@ -470,9 +470,9 @@ def send_request(method, path, params, options)
           end
           case error
             when Faraday::TimeoutError
-              raise Ably::Exceptions::ConnectionTimeout.new(error.message, nil, 80014, error, {}, random_request_id)
+              raise Ably::Exceptions::ConnectionTimeout.new(error.message, nil, 80014, error, {request_id: random_request_id})
```

Comment:
Spacing issues again, here and many times below

Reply:
I am not a big fan of this. `random_request_id` is defined in line 445, yet is within a conditional.  I realise Ruby allows this sort of thing, but it's messy and error prone.  Why are you defining `random_request_id` as a varaiable when you have access to the `params` variable in this scope which is available anyway?

Reply:
In fact, I am not sure I have seen a test that confirms that a) a request is sent to Ably with a `request_id`, b) it fails (intentionally), c) the test confirms that the exception raised has that SAME `request_id`, d) the Logger is showing the `request_id` in the output.  This is the crux of what we're doing here.  Given this code path, I fear our tests so far are just testing that a `request_id` is present, but not ensuring it is consistent throughout.

Reply:
This makes sense, but I am struggling a bit to inspect the url of a request, say `client.stats`. Any suggestion?